### PR TITLE
Dropped support for Symfony 2, added support for Symfony 4. Fixed most PHP warnings.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: Security/RoleAndIpStrategy.php [hunk 1]
```diff
@@ -47,8 +52,8 @@ public function isAllowed(Request $request)
             return false;
         }
 
-        $availableRoles = array_map(function (RoleInterface $role) {
-            return $role->getRole();
+        $availableRoles = array_map(function ($role) {
+            return (string)$role;
```
Comment (mariusbalcytis) [inline Security/RoleAndIpStrategy.php]: No conversion to string available here – see failed tests

File: composer.json [hunk 2]
```diff
@@ -7,16 +7,22 @@
...
-        "php": ">=5.5",
-        "symfony/symfony": "^2.6|3.*",
-        "paysera/lib-serializer": "^2.0.2",
-        "doctrine/orm": "^2.0"
-    },
-    "require-dev": {
-        "phpunit/phpunit": "^4.8",
-        "mockery/mockery": "^0.9.4"
+        "php": "^7.0",
+        "paysera/lib-serializer": "^3.0",
```
Comment (mariusbalcytis) [inline composer.json:11]: Is 2.* support really dropped? It's better to leave `^2.6|^3.0` if it's still supported

**Review by mariusbalcytis — CHANGES_REQUESTED** [review]

File: composer.json (hunk 2 above)
Comment (ghost) [reply composer.json:11]: Why bother with Symfony 2? I've removed support for it in these changes.

File: .travis.yml [hunk 3]
```diff
@@ -7,8 +7,6 @@ env:
     - COMPOSER_NO_INTERACTION="1"
 
 php:
```
Comment (mariusbalcytis) [inline .travis.yml]: Let's include the test for 3.x version:
```
matrix:
    include:
      - php: 7.0
        env: COMPOSER_ARGS=""
      - php: 7.1
        env: COMPOSER_ARGS=""
      - php: 7.2
        env: COMPOSER_ARGS=""

      - php: 7.0
        env: COMPOSER_ARGS="--prefer-lowest"
      - php: 7.1
        env: COMPOSER_ARGS="--prefer-lowest"
      - php: 7.2
        env: COMPOSER_ARGS="--prefer-lowest"

cache:
    directories:
      - $HOME/.composer/cache

before_script:
    - phpenv config-rm xdebug.ini || true
    - travis_retry composer update ${COMPOSER_FLAGS} --no-interaction --prefer-dist
```

Comment (ghost) [reply .travis.yml]: Nice

**Review by mariusbalcytis — APPROVED** [review]

**Pull request merged**